src/machines/authMachine.ts

```diff
@@ -108,18 +111,16 @@ export const authMachine = Machine<AuthMachineContext, AuthMachineSchema, AuthMa
             throw new Error("Username or password is invalid");
           });
       },
-      getUserProfile: async (ctx, event) => {
-        const resp = await httpClient.get(`http://localhost:3001/checkAuth`);
-        return resp.data;
-      },
+      getUserProfile: async (ctx, event) =>
+        await httpClient.get(`http://localhost:3001/checkAuth`).then(({ data }) => data),
```

Comment:
Any reason for chaining a `.then` instead of destructing `data` and returning similar to how you've done it in other actions?